packages/app/src/components/DetailsBox.tsx

```diff
@@ -0,0 +1,139 @@
+import clx from "classnames";
+import { FC, useState } from "react";
+import { useSunriseStake } from "../context/sunriseStakeContext";
+import { toFixedWithPrecision, toSol, ZERO } from "../lib/util";
+import BN from "bn.js";
+import { ChevronUpIcon } from "@heroicons/react/20/solid";
+
+interface DetailEntryProps {
+  label: string;
+  value: string;
+  share?: number;
+}
+const DetailEntry: FC<DetailEntryProps> = ({ label, value, share }) => (
+  <div className="flex flex-row justify-between">
+    <div className="text-gray-100 text-sm sm:text-lg">{label}</div>
+    <div className="font-bold text-sm sm:text-lg">
+      {value} <span className="text-bold text-xs">SOL</span>{" "}
+      {share !== undefined && (
+        <span className="hidden sm:inline text-gray-300 font-normal text-sm">
+          ({share}%)
+        </span>
+      )}
+    </div>
+  </div>
+);
+
+interface Props {
+  className?: string;
+}
+const DetailsBox: FC<Props> = ({ className }) => {
+  const { details } = useSunriseStake();
+  const [isShowing, setIsShowing] = useState(false);
+
+  if (!details) return <>Loading...</>;
+
+  const inflightTotal = details.inflight.reduce(
+    (acc, x) => acc.add(x.totalOrderedLamports),
+    ZERO
+  );
+
+  const totalValue = details.mpDetails.msolValue
+    .add(details.lpDetails.lpSolValue)
+    .add(details.bpDetails.bsolValue)
+    .add(inflightTotal);
+
+  const mpShare =
+    details.mpDetails.msolValue.muln(1_000).div(totalValue).toNumber() / 10;
+  const bpShare =
+    details.bpDetails.bsolValue.muln(1_000).div(totalValue).toNumber() / 10;
+  const lpShare =
+    details.lpDetails.lpSolValue.muln(1_000).div(totalValue).toNumber() / 10;
+  const inflightShare =
+    inflightTotal.muln(1_000).div(totalValue).toNumber() / 10;
+
+  const extractableYield = new BN(
+    Math.max(details.extractableYield.toNumber(), 0)
+  );
+  const yieldShare =
+    extractableYield.muln(1_000).div(totalValue).toNumber() / 10;
+  const gSolSupply = new BN(details.balances.gsolSupply.amount);
+
+  const lamportsToDisplay = (lamports: BN): string =>
+    toFixedWithPrecision(toSol(lamports), 2);
+
+  return (
+    <>
+      <button
+        onClick={() => setIsShowing((isShowing) => !isShowing)}
+        className={clx(
+          "transition duration-700 flex w-full justify-between rounded-t-md px-4 py-1 text-left text-sm font-medium text-white ",
+          {
+            "rounded-t-md backdrop-blur-sm": isShowing,
+            "rounded-md bg-inset": !isShowing,
+          }
+        )}
+      >
+        <span>Details</span>
+        <ChevronUpIcon
```

Comment:
I think these should be switched - ChevronUp when open, ChevronDown when closed.

Reply:
Done